Sources/AWSLambdaRuntimeCore/LambdaRunner.swift

```diff
@@ -80,13 +89,18 @@ extension Lambda {
                         if case .failure(let error) = result {
                             logger.warning("lambda handler returned an error: \(error)")
                         }
-                        return (invocation, result)
+                        return (invocation, result, baggage)
                     }
-            }.flatMap { invocation, result in
+            }.flatMap { (invocation, result, baggage: BaggageContext) in
```

Comment:
I'd pass the context not just baggage. seems cleaner

Reply:
`BaggageContext` is value type, `Lambda.Context` is reference type

here we want to pass the "original" baggage value, while the value of `context.baggage` may have been changed by the time we get the handler result

maybe `Lambda.Context` should also be a struct or perhaps `BaggageContext` should be propagated separately?

@ktoso WDYT?